starport/pkg/cosmosgen/templates/vuex/store/index.ts.tpl

```diff
@@ -92,55 +103,70 @@ export default {
 				dispatch(subscription.action, subscription.payload)
 			})
 		},
-		{{ range .Module.HTTPQueries }}async {{ .FullName }}({ commit, rootGetters, getters }, { options: { subscribe = false , all = false}, params: {...key}, query=null }) {
+		{{ range .Module.HTTPQueries }}async {{ .FullName }}({ commit, rootGetters, getters }, { options: { subscribe, all} = { subscribe:false, all:false}, params: {...key}, query=null }) {
 			try {
-				
-				let value = query?(await (await initQueryClient(rootGetters)).{{ camelCase .FullName }}({{ range $i,$a :=(index .Rules 0).Params}} key.{{$a}}, {{end}} query)).data:(await (await initQueryClient(rootGetters)).{{ camelCase .FullName }}({{ range $i,$a :=(index .Rules 0).Params}}{{ if (gt $i 0)}}, {{ end}} key.{{$a}} {{end}})).data
+				const queryClient=await initQueryClient(rootGetters)
+				let value
+				{{ if (index .Rules 0).HasBody}}
```

Comment:
https://github.com/tendermint/starport/blob/a92af9dcf1728fff49cabe1374703d488c0027c4/starport/pkg/cosmosgen/templates/vuex/store/index.ts.tpl#L110-L122

This whole section above can be simplified to a single line, something similar to below:

```
value = (await queryClient.{{ camelCase .FullName }}({{ range $i,$a := .Rule.Params}} key.{{$a}}, {{end}} {{ >> IF QUERY ADD QUERY }} {{ >> IF BODY ADD BODY }} )).data
```

Reply:
I have a vague recollection there was a reason for the second part with the `if (query)` in JS....But I cant see it now...maybe its no longer needed due to some other refactoring elsewhere so I'll change it.

Reply:
Can you please use `additional_bindings` for one of the RPC funcs of your choosing and run the `starport build` and check the generated `rest.js` for it?

You'll see a function being generated with the same name but each has a different suffix (`index+1`).

https://github.com/googleapis/googleapis/blob/ca1372c6d7bcb199638ebfdb40d2b2660bab7b88/google/api/http.proto#L182-L199